src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.VsLanguageDebugInfo.cs

```diff
@@ -92,21 +92,25 @@ public int GetNameOfLocation(IVsTextBuffer pBuffer, int iLine, int iCol, out str
                             var textBuffer = _languageService.EditorAdaptersFactoryService.GetDataBuffer(pBuffer);
                             if (textBuffer != null)
                             {
-                                var point = textBuffer.CurrentSnapshot.GetPoint(iLine, iCol);
-                                var document = point.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
-
-                                if (document != null)
+                                var nullablePoint = textBuffer.CurrentSnapshot.TryGetPoint(iLine, iCol);
+                                if (nullablePoint.HasValue)
                                 {
-                                    // NOTE(cyrusn): We have to wait here because the debuggers' 
-                                    // GetNameOfLocation is a blocking call.  In the future, it 
-                                    // would be nice if they could make it async.
-                                    var debugLocationInfo = _languageDebugInfo.GetLocationInfoAsync(document, point, cancellationToken).WaitAndGetResult(cancellationToken);
+                                    var point = nullablePoint.Value;
+                                    var document = point.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
 
-                                    if (!debugLocationInfo.IsDefault)
+                                    if (document != null)
                                     {
-                                        succeeded = true;
-                                        name = debugLocationInfo.Name;
-                                        lineOffset = debugLocationInfo.LineOffset;
+                                        // NOTE(cyrusn): We have to wait here because the debuggers' 
+                                        // GetNameOfLocation is a blocking call.  In the future, it 
+                                        // would be nice if they could make it async.
+                                        var debugLocationInfo = _languageDebugInfo.GetLocationInfoAsync(document, point, cancellationToken).WaitAndGetResult(cancellationToken);
+
+                                        if (!debugLocationInfo.IsDefault)
+                                        {
+                                            succeeded = true;
+                                            name = debugLocationInfo.Name;
+                                            lineOffset = debugLocationInfo.LineOffset;
+                                        }
                                     }
                                 }
                             }
```

Comment:
If we extracted this to a method, could we write a unit test for such method?

Reply:
The method would take textBuffer, lineNumber and columnNumber

---
In reply to: [181254431](https://github.com/dotnet/roslyn/pull/26128#discussion_r181254431) [](ancestors = 181254431)

Reply:
agreed to take this after 15.7
added [https://github.com/dotnet/roslyn/issues/26133](https://github.com/dotnet/roslyn/issues/26133)

---
In reply to: [181254629](https://github.com/dotnet/roslyn/pull/26128#discussion_r181254629) [](ancestors = 181254629,181254431)